src/hotspot/cpu/riscv/riscv_v.ad

```diff
@@ -2878,6 +2878,58 @@ instruct vclearArray_reg_reg(iRegL_R29 cnt, iRegP_R28 base, Universe dummy,
   ins_pipe(pipe_class_memory);
 %}
 
+// CompressBits of Long & Integer
+
+instruct compressBitsI(iRegINoSp dst, iRegIorL2I src, iRegIorL2I mask, iRegPNoSp tmp, vRegMask_V0 v0, vReg_V4 v4, vReg_V8 v8) %{
+  predicate(UseRVV && (MaxVectorSize >= 16));
+  match(Set dst (CompressBits src mask));
+  effect(TEMP tmp, TEMP v0, TEMP v4, TEMP v8);
+  format %{ "vsetivli x0, 1, e32, m2, tu, mu\t#@compressBitsI\n\t"
+            "vmv.s.x $v0, $src\n\t"
+            "mv $tmp, 32\n\t"
+            "vsetvli x0, $tmp, e8, m2, tu, mu\n\t"
+            "vmv.v.i $v4, 0\n\t"
+            "vmerge.vim $v4, $v4, 1, $v0\n\t"
+            "vmv.v.i $v8, 0\n\t"
+            "vsetivli x0, 1, e32, m2, tu, mu\n\t"
+            "vmv.v.x $v0, $mask\n\t"
+            "vsetvli x0, $tmp, e8, m2, tu, mu\n\t"
+            "vcompress.vm $v8, $v4, $v0\n\t"
+            "vmseq.vi $v0, $v8, 1\n\t"
+            "vsetivli x0, 1, e32, m2, tu, mu\n\t"
+            "vmv.x.s $dst, $v0\t#@compressBitsI\n\t"
+          %}
+  ins_encode %{
+    __ compress_bits_i_v(as_Register($dst$$reg), as_Register($src$$reg), as_Register($mask$$reg), as_Register($tmp$$reg));
+  %}
+  ins_pipe(pipe_slow);
+%}
+
+instruct compressBitsL(iRegLNoSp dst, iRegL src, iRegL mask, iRegPNoSp tmp, vRegMask_V0 v0, vReg_V4 v4, vReg_V8 v8) %{
+  predicate(UseRVV && (MaxVectorSize >= 16));
```

Comment:
Yes, seems it's redundant to check it here again, as we already have a check in Matcher::match_rule_supported where it's necessary to avoid silent failure in case restrictions on MaxVectorSize is changed in the future.
It's removed here.